packages/purpose-process/src/services/purposeService.ts

```diff
@@ -636,6 +658,87 @@ export function purposeServiceBuilder(
       );
       return { purpose, isRiskAnalysisValid: validatedFormSeed !== undefined };
     },
+    async createReversePurpose(
+      organizationId: TenantId,
+      seed: ApiReversePurposeSeed,
+      correlationId: string,
+      logger: Logger
+    ): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
+      logger.info(
+        `Creating Purpose for EService ${seed.eServiceId}, Consumer ${seed.consumerId}`
+      );
+      const eserviceId: EServiceId = unsafeBrandId(seed.eServiceId);
+      const consumerId: TenantId = unsafeBrandId(seed.consumerId);
+
+      assertOrganizationIsAConsumer(organizationId, consumerId);
+      const eservice = await retrieveEService(eserviceId, readModelService);
+      assertEserviceMode(eservice, eserviceMode.receive);
+
+      const riskAnalysis = retrieveRiskAnalysis(
+        unsafeBrandId(seed.riskAnalysisId),
+        eservice
+      );
+
+      assertConsistentFreeOfCharge(
+        seed.isFreeOfCharge,
+        seed.freeOfChargeReason
+      );
+
+      const producer = await retrieveTenant(
+        eservice.producerId,
+        readModelService
+      );
+
+      assertTenantKindExists(producer);
+
+      await retrieveActiveAgreement(eserviceId, consumerId, readModelService);
+
+      const purposeWithSameName = await readModelService.getPurpose(
+        eserviceId,
+        consumerId,
+        seed.title
+      );
+
+      if (purposeWithSameName) {
+        throw duplicatedPurposeTitle(seed.title);
+      }
+
+      validateRiskAnalysisOrThrow(
+        riskAnalysisFormToRiskAnalysisFormToValidate(
+          riskAnalysis.riskAnalysisForm
+        ),
+        false,
```

Comment:
minor point: understand what the boolean refers to is a bit difficult, it could be worth having a named parameter.
Same, not mandatory, just nice

Reply:
Updated